src/thor/dijkstras.cc

```diff
@@ -622,31 +627,20 @@ void Dijkstras::ExpandForwardMultiModal(GraphReader& graphreader,
     newcost += transition_cost;
 
     // Prohibit entering the same station as the prior.
-    if (directededge->use() == Use::kTransitConnection &&
+    // TODO: if pred was a station, we wouldn't know about it; we only set
+    // prior_stopid for platforms so far; also pull this check much further up
+    if (directededge->use() == Use::kPlatformConnection &&
         directededge->endnode() == pred.prior_stopid()) {
       continue;
     }
 
     // Test if exceeding maximum transfer walking distance
-    if (directededge->use() == Use::kTransitConnection && pred.prior_stopid().Is_Valid() &&
+    // TODO: transfer distance != walking distance! (one more label member?)
+    if (directededge->use() == Use::kPlatformConnection && pred.prior_stopid().Is_Valid() &&
         walking_distance > max_transfer_distance_) {
       continue;
     }
```

Comment:
these ones are not really proper IMO, same situation as in the routing algo (where I updated it too with comments). let's reason over this together tmrw